test/integration/Test/Integration/Scenario/Wallets.hs

```diff
@@ -28,33 +29,108 @@ import Test.Integration.Framework.DSL
     , expectFieldEqual
     , expectFieldNotEqual
+    , expectListItemFieldEqual
+    , expectListSizeEqual
     , expectResponseCode
     , json
     , passphraseLastUpdate
     , request
     , state
+    , verify
     , walletId
     , walletName
     )
 
+import qualified Network.HTTP.Types.Status as HTTP
+
 spec :: SpecWith Context
 spec = do
-    it "Create a wallet" $ \ctx -> do
+    it "WALLETS_CREATE_01 - Create a wallet" $ \ctx -> do
 
         let payload = Json [json| {
                 "name": "1st Wallet",
-                "mnemonic_sentence": ["identify", "screen", "lock", "bargain", "inch", "drop", "canyon", "flock", "dry", "zone", "wash", "argue", "system", "glory", "light"],
-                "mnemonic_second_factor": ["attract", "tornado", "slender", "pumpkin", "clown", "announce", "term", "winner", "ready"],
+                "mnemonic_sentence": #{mnemonics15},
+                "mnemonic_second_factor": #{mnemonics12},
                 "passphrase": "Secure Passphrase",
                 "address_pool_gap": 30
                 } |]
         r <- request @ApiWallet ctx ("POST", "v2/wallets") Default payload
-        expectResponseCode @IO HTTP.status202 r
-        expectFieldEqual walletName "1st Wallet" r
-        expectFieldEqual addressPoolGap 30 r
-        expectFieldEqual balanceAvailable 0 r
-        expectFieldEqual balanceTotal 0 r
-        expectFieldEqual delegation (NotDelegating) r
-        expectFieldEqual state (Restoring (Quantity minBound)) r
-        expectFieldEqual walletId "4bba596017a63e52a4cf6caaadf8a670bfe4745b" r
-        expectFieldNotEqual passphraseLastUpdate "2019-04-12 07:57:28.439742724 UTC" r
+        verify r
+            [ expectResponseCode @IO HTTP.status202
+            , expectFieldEqual walletName "1st Wallet"
+            , expectFieldEqual addressPoolGap 30
+            , expectFieldEqual balanceAvailable 0
+            , expectFieldEqual balanceTotal 0
+            , expectFieldEqual state (Restoring (Quantity minBound))
+            , expectFieldEqual delegation (NotDelegating)
+            , expectFieldEqual walletId "2cf060fe53e4e0593f145f22b858dfc60676d4ab"
+            , expectFieldNotEqual passphraseLastUpdate "2019-04-12 07:57:28.439742724 UTC"
+            ]
+
+    it "WALLETS_CREATE_01 - Created a wallet can be listed" $ \ctx -> do
+
+        let payload = Json [json| {
+                "name": "Wallet to be listed",
+                "mnemonic_sentence": #{mnemonics18},
+                "mnemonic_second_factor": #{mnemonics9},
+                "passphrase": "Secure Passphrase",
+                "address_pool_gap": 20
+                } |]
+        _ <- request @ApiWallet ctx ("POST", "v2/wallets") Default payload
+        r2 <- request @[ApiWallet] ctx ("GET", "v2/wallets") Default Empty
+        verify r2
+            [ expectResponseCode @IO HTTP.status200
+            , expectListSizeEqual 1
+            , expectListItemFieldEqual 0 walletName "Wallet to be listed"
```

Comment:
I wonder here if we couldn't just favor lense composition over having a new accessor. Something like:

```hs
expectFiedEqual (ix 0 . walletName) "Wallet to be listed"
```

Reply:
I'm sure that's possible (since you bring it up;-) ). However, the particular one above,  does not work for me... :)

For now I'd be inclined to merge as is and maybe spend some time on that later on (on some subsequent PR), cool?

Reply:
Yup, that was really just a small suggestion :)